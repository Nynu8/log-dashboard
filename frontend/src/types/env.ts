export type Env = 'Prd' | 'Acc' | 'Dev';
