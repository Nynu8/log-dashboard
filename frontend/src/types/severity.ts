export type Severity = 'Info' | 'Debug' | 'Warning' | 'Error';
