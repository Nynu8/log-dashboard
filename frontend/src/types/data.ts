import { Env } from './env';
import { Severity } from './severity';

export interface LogEntry {
  id: string;
  env: Env;
  timestamp: string;
  severity: Severity;
  source: string;
  logData: string;
}

export type Data = LogEntry[];
