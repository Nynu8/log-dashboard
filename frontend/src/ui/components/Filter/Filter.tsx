import { Dispatch, SetStateAction } from 'react';
import { Divider, FormControlLabel, Radio, RadioGroup } from '@mui/material';
import { LogEntry } from 'types/data';
import { Env } from 'types/env';
import { Severity } from 'types/severity';

export interface Filters {
  env: Env | '';
  severity: Severity | '';
  source: string;
}

export interface FilterProps {
  setFilters: Dispatch<SetStateAction<Filters>>;
  filters: Filters;
  data: LogEntry[];
}

export default function Filter({ data, filters, setFilters }: FilterProps) {
  const handleEnvClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const { value } = event.target as HTMLInputElement;

    if (value === filters.env) {
      setFilters({
        ...filters,
        env: '',
      });
    } else {
      setFilters({
        ...filters,
        env: value as Env,
      });
    }
  };

  const handleSeverityClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const { value } = event.target as HTMLInputElement;

    if (value === filters.severity) {
      setFilters({
        ...filters,
        severity: '',
      });
    } else {
      setFilters({
        ...filters,
        severity: value as Severity,
      });
    }
  };

  const handleSourceClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const { value } = event.target as HTMLInputElement;

    if (value === filters.source) {
      setFilters({
        ...filters,
        source: '',
      });
    } else {
      setFilters({
        ...filters,
        source: value,
      });
    }
  };

  const getUniqueSources = () => [...new Set(data.map((log) => log.source))];

  return (
    <>
      <h2>Environment</h2>
      <RadioGroup name="env" value={filters.env ?? ''}>
        <FormControlLabel value="Dev" control={<Radio onClick={handleEnvClick} />} label="Development" />
        <FormControlLabel value="Acc" control={<Radio onClick={handleEnvClick} />} label="Acceptance" />
        <FormControlLabel value="Prd" control={<Radio onClick={handleEnvClick} />} label="Production" />
      </RadioGroup>
      <RadioGroup name="source"></RadioGroup>
      <Divider />
      <h2>Severity</h2>
      <RadioGroup name="severity" value={filters.severity ?? ''}>
        <FormControlLabel value="Error" control={<Radio onClick={handleSeverityClick} />} label="Error" />
        <FormControlLabel value="Warning" control={<Radio onClick={handleSeverityClick} />} label="Warning" />
        <FormControlLabel value="Info" control={<Radio onClick={handleSeverityClick} />} label="Info" />
        <FormControlLabel value="Debug" control={<Radio onClick={handleSeverityClick} />} label="Debug" />
      </RadioGroup>
      <RadioGroup name="source"></RadioGroup>
      {getUniqueSources().length > 0 && (
        <>
          <Divider />
          <h2>Source</h2>
          <RadioGroup name="source" value={filters.source ?? ''}>
            {getUniqueSources().map((source) => (
              <FormControlLabel
                value={source}
                control={<Radio onClick={handleSourceClick} />}
                label={source}
                key={source}
              />
            ))}
          </RadioGroup>
        </>
      )}
    </>
  );
}
