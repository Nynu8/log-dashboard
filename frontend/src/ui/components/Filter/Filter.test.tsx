import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Filter, { Filters } from './Filter';
import { LogEntry } from 'types/data';

const filters = {} as Filters;
const setFilters = jest.fn();

test('has empty sources when data is empty', () => {
  const data = [] as LogEntry[];
  render(<Filter filters={filters} setFilters={setFilters} data={data} />);
  const element = screen.queryByText(/source/i);

  expect(element).not.toBeInTheDocument();
});

test('has correct sources when data exists', () => {
  const data = [{ source: 'a' }, { source: 'b' }, { source: 'c' }] as LogEntry[];
  const component = renderer.create(<Filter filters={filters} setFilters={setFilters} data={data} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
