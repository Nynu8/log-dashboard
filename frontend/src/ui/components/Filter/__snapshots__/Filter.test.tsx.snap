// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has correct sources when data exists 1`] = `
Array [
  <h2>
    Environment
  </h2>,
  <div
    className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    role="radiogroup"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="env"
          onChange={[Function]}
          type="radio"
          value="Dev"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Development
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="env"
          onChange={[Function]}
          type="radio"
          value="Acc"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Acceptance
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="env"
          onChange={[Function]}
          type="radio"
          value="Prd"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Production
      </span>
    </label>
  </div>,
  <div
    className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    role="radiogroup"
  />,
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />,
  <h2>
    Severity
  </h2>,
  <div
    className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    role="radiogroup"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="severity"
          onChange={[Function]}
          type="radio"
          value="Error"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Error
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="severity"
          onChange={[Function]}
          type="radio"
          value="Warning"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Warning
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="severity"
          onChange={[Function]}
          type="radio"
          value="Info"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Info
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="severity"
          onChange={[Function]}
          type="radio"
          value="Debug"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Debug
      </span>
    </label>
  </div>,
  <div
    className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    role="radiogroup"
  />,
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />,
  <h2>
    Source
  </h2>,
  <div
    className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    role="radiogroup"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="source"
          onChange={[Function]}
          type="radio"
          value="a"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        a
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="source"
          onChange={[Function]}
          type="radio"
          value="b"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        b
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="source"
          onChange={[Function]}
          type="radio"
          value="c"
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        c
      </span>
    </label>
  </div>,
]
`;
